Sources/ClientRuntime/Networking/Streaming/BufferedStream.swift

```diff
@@ -32,6 +32,10 @@ public class BufferedStream: Stream {
     private(set) var buffer: Data?
     private let lock = NSRecursiveLock()
 
+    /// Continuations that are awaiting data, and the byte count to be delivered to each.
+    /// Access to this value should only occur while `lock` is locked.
+    private var clients: [(UnsafeContinuation<Data?, Error>, Int)] = []
```

Comment:
IMO `clients` is an overloaded term, perhaps use `callers` or `receivers` instead.

Reply:
Can we ever have multiple concurrent clients? Should this be a single item such as `(UnsafeContinuation<Data?, Error>, Int)?` instead of an array?

Reply:
`readers` is better, I'll use that.

Also, what would be the behavior if we don't accept multiple suspended readers?  Fail with an error if there is already a reader suspended?  Return nil or empty data to the existing suspended reader then replace it with the new one?  Maintaining a queue sidesteps this question.